File: snippets/nodejs/typing_animation.js
// Typing Animation in Terminal with Nodejs
function typeEffect(message, delay = 50) {
  return new Promise(async (resolve) => {
    for (let i = 0; i < message.length; i++) {
      process.stdout.write(message[i]);
      await new Promise(r => setTimeout(r, delay));
    }
    process.stdout.write('\n');
    resolve();
  });
}

// Example
async function main() {
  await typeEffect("Hello, this is a typing animation...");
}

main();

File: snippets/python/typing_animation.py
import sys
import time

def type_effect(text, delay=0.05):

    for char in text:
        sys.stdout.write(char)
        sys.stdout.flush()
        time.sleep(delay)
    print()

if __name__ == "__main__":
    type_effect("Hello, this is a typing animation in Python!")
    type_effect("Now the next line runs after the first is done!", 0.03)
    print("✅ All animations completed!")
